modules/ts/src/ts_func.cpp

```diff
@@ -2,6 +2,10 @@
 #include <float.h>
 #include <limits.h>
 #include "opencv2/imgproc/types_c.h"
+#include "../src/utils/depth_dispatch.hpp"
```

Comment:
It makes sense to put this header into `core/include/` instead of `core/src/`

Reply:
Can you help me identify the correct naming/subfolder, so that this header doesn't leak to the public API?

Reply:
> leak to the public API

What do you mean?
- documentation? - look for `//! @cond IGNORED` for the whole file
- install directory? - AFAIK, "private/" handles that.

Implementation itself should be marked as "static inline" (it is template anyway) and may be placed into `cv::detail` namespace.

---

- "core/utils/" - contains low level / system utility functions (not a good place)
- "core/detail/" - good candidate
- just "opencv2/core/" should work too (but don't include that into core.hpp or base.hpp)

Preferable name is "dispatch_helper.impl.hpp".